pylint/lint/utils.py

```diff
@@ -5,14 +5,32 @@
 from __future__ import annotations
 
 import contextlib
+import os
 import sys
 import traceback
 from collections.abc import Iterator, Sequence
 from datetime import datetime
 from pathlib import Path
 
 from pylint.config import PYLINT_HOME
-from pylint.lint.expand_modules import get_python_path
+
+
+def get_python_path(filepath: str) -> str:
+    """TODO This get the python path with the (bad) assumption that there is always
+    an __init__.py.
+
+    This is not true since python 3.3 and is causing problem.
+    """
+    dirname = os.path.realpath(os.path.expanduser(filepath))
+    if not os.path.isdir(dirname):
+        dirname = os.path.dirname(dirname)
+    while True:
+        if not os.path.exists(os.path.join(dirname, "__init__.py")):
+            return dirname
+        old_dirname = dirname
+        dirname = os.path.dirname(dirname)
+        if old_dirname == dirname:
+            return os.getcwd()
```

Comment:
This condition is never true during the tests, could we add a test ?

Reply:
This is currently uncovered right? I'd like the investigate the TODO in the comment at some point, but for now I think focusing on the issue at the hand is big enough. I'm kind of worried about the effect of now warning on the correct module for users who rely on the false module right now.

Reply:
Fair enough !